include/status-code/status_code.hpp

```diff
@@ -379,8 +394,37 @@ namespace detail
     {
     }
   };
+
+  template <class DomainType> struct has_stateful_mixin
+  {
+    static constexpr bool value = sizeof(status_code_storage<DomainType>) != sizeof(mixins::mixin<status_code_storage<DomainType>, DomainType>);
+  };
+  template <> struct has_stateful_mixin<void>
+  {
+    static constexpr bool value = false;
+  };
+
+  template <class FromDomain, class ToValueType> struct is_type_erasable_to
+  {
+    static constexpr bool value = traits::is_move_bitcopying<typename status_code_storage<FromDomain>::value_type>::value         //
+                                  && sizeof(status_code_storage<FromDomain>) <= sizeof(status_code_storage<erased<ToValueType>>)  //
+                                  && !has_stateful_mixin<FromDomain>::value;
+  };
+  template <class ToValueType> struct is_type_erasable_to<void, ToValueType>
+  {
+    static constexpr bool value = false;
+  };
 }  // namespace detail
 
+namespace traits
+{
+  //! Determines whether the mixin contained in `StatusCode` contains non-static member variables.
+  template <class StatusCode> using has_stateful_mixin = detail::has_stateful_mixin<typename detail::remove_cvref<StatusCode>::type::value_type>;
+
+  //! Determines whether the status code `From` can be type erased into a status code based on `ToValueType`.
+  template <class From, class ToValueType> using is_type_erasable_to = detail::is_type_erasable_to<typename detail::remove_cvref<From>::type::domain_type, ToValueType>;
```

Comment:
Should the trait be SFINAE friendly for non `status_code` `From` type arguments?

Reply:
Surely not, as it only gets called from a requires context and therefore if it's invalid code the requires is no?

I say "surely" here because for enable_if this is definitely true, whereas for Concept requires the evaluation is somewhat earlier in the evaluation cycle and invalid code can error. OTOH I'm happy to leave it as currently is and if it fails in production it can be fixed later.